src/Analyzers/CSharp/Tests/UseParameterNullChecking/UseParameterNullCheckingTests.cs

```diff
@@ -0,0 +1,1416 @@
+    public C(string s!!)
+    {
+        [|if (s is null)
+            throw new ArgumentNullException(nameof(s));|]
+    }
+}",
+                FixedCode = @"using System;
+
+class C
+{
+    public C(string s!!)
+    {
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestSingleLineNoBraces()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"using System;
+
+class C
+{
+    public C(string s)
+    {
+        [|if (s is null) throw new ArgumentNullException(nameof(s));|]
+    }
+}",
+                FixedCode = @"using System;
+
+class C
+{
+    public C(string s!!)
+    {
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestSingleLineWithBraces()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"using System;
+
+class C
+{
+    public C(string s)
+    {
+        [|if (s is null) { throw new ArgumentNullException(nameof(s)); }|]
+    }
+}",
+                FixedCode = @"using System;
+
+class C
+{
+    public C(string s!!)
+    {
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestLeadingAndTrailingTrivia()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"using System;
+
+class C
+{
+    public C(string s)
+    {
+        // comment1
+        [|if (s is null) { throw new ArgumentNullException(nameof(s)); }|] // comment2
+    }
+}",
+                FixedCode = @"using System;
+
+class C
+{
+    public C(string s!!)
+    {
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestAssignThenTest()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"using System;
+
+class C
+{
+    public C(string s)
+    {
+        s = ""a"";
+        [|if (s is null) { throw new ArgumentNullException(nameof(s)); }|]
+    }
+}",
+                FixedCode = @"using System;
+
+class C
+{
+    public C(string s!!)
+    {
+        s = ""a"";
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestRedundantNullChecks()
+        {
+            // TODO: what test framework settings do we use here?
```

Comment:
@sharwell I am not certain how to configure the test framework to handle this scenario. When I run this test in its current state I get the following result:

```log
Context: Iterative code fix application
content of '/0/Test0.cs' did not match. Diff shown with expected as baseline:
 
 using System;
 
 class C
 {
     public C(string s!!)
     {
-        if (s is null)
-            throw new ArgumentNullException(nameof(s));
     }
 }

Expected: True
Actual:   False
```

Basically, I would like to just demonstrate that the analyzer/fixer does not do something bad in this scenario, such as crash or produce erroneous code. I don't think it's important to eliminate both marked if-statements in a single iteration of the fixer, for example.

Reply:
The test runs iterations until there is nothing left to fix. It will fail if _any_ iteration causes a problem.

Reply:
Thanks again for the guidance. I think the test is now fixed.